### PR TITLE
[TransactionRequestInput] - add fields

### PR DESCRIPTION
**Pull request by fabiorigam**
## Summary by CodeRabbit

- **New Features**
	- Introduced optional `delegationUrl` and `comment` properties to transaction request inputs and contract transaction options, enhancing transaction metadata.
	- Updated transaction execution methods to accept new parameters for delegation and comments.

- **Bug Fixes**
	- Enhanced error handling in the transaction signing process.

- **Tests**
	- Added new test cases for signing transactions with delegation URLs and ERC20 contract functionality on the testnet.

### CODE REVIEW

**Comment by claytonneal** [comment]
@coderabbitai summary

**Comment by claytonneal** [comment]
@coderabbitai review

File: packages/network/tests/signer/signers/vechain-private-key-signer/vechain-private-key-signer.testnet.test.ts [hunk 1]
```diff
@@ -102,6 +102,32 @@ describe('VeChain base signer tests - testnet', () => {
...
+                            thorClient,
+                            THOR_SOLO_ACCOUNTS_BASE_WALLET,
+                            false
+                        )
+                    );
+
+                    // Sign the transaction
+                    const signedTransaction = await signer.signTransaction({
+                        from: fixture.origin.address,
+                        delegationUrl: fixture.options.delegatorUrl
```
Comment (leszek-vechain) [inline packages/network/tests/signer/signers/vechain-private-key-signer/vechain-private-key-signer.testnet.test.ts:124]: can we add some negative tests ? like invalid url etc ?

Comment (fabiorigam) [reply packages/network/tests/signer/signers/vechain-private-key-signer/vechain-private-key-signer.testnet.test.ts:124]: I'll add a ticket for that

**Review by leszek-vechain — APPROVED** [review]

**Pull request merged**